### ISSUE TITLE
Non-integer unit exponents?

### ISSUE THREAD

**Issue by MichaelClerx** [ModellingWebLab/cellmlmanip#48]
Are we allowing this:
```
x = 5 m
y = sqrt(x)
```
?

and what about this:
```
a = 5 m
b = a ^ 2.345
```
?

**Comment by MichaelClerx** [comment]
(This is related to the issue of determining the units of `a ^ b`)

**Comment by MichaelClerx** [comment]
In Myokit I've explicitly disallowed it, with zero errors and/or complaints so far, but I can imagine you might end up with them somehow?

**Comment by jonc125** [comment]
Strictly speaking the first is invalid (sqrt requires dimensionless args, similarly for exp) and pycml emits an error. For web lab code generation, we treat such errors just as warnings and proceed hoping the modeller put conversion factors in appropriately. Since some widely used models have these issues...

The second option is allowed. But hopefully not done much!

**Comment by MichaelClerx** [comment]
Really? So we allow m^(1/2) but not sqrt(m)? I would have thought having a rational in the exponent would be permissible before having a real was!

**Comment by MichaelClerx** [comment]
Also sqrt(m^2) should be absolutely fine. For example if you measure the area of a square and want to deduce its length. I would say the input of sqrt() can have a unit, as long as its output doesn't have a non-integer unit...

**Comment by jonc125** [comment]
Good point, sorry, getting confused there. It is an error for exp() and trig functions, but not roots as you say.
